operators/src/source/ogr_source.rs

```diff
@@ -0,0 +1,2856 @@
+            let empty_collection = feature_collection_builder
+                .finish_header()
+                .build()
+                .map_err(Into::into);
+
+            if poll_result_sender
+                .send(Some(empty_collection))
+                .and_then(|_| poll_result_sender.send(None))
+                .is_ok()
+            {
+                work_query.waker.wake_by_ref();
+            }
+
+            return Ok(());
+        }
+
+        while features.peek().is_some() {
+            let batch_result = Self::compute_batch(
+                &mut features,
+                feature_collection_builder.clone(),
+                dataset_information,
+                &default_geometry,
+                &data_types,
+                query_rectangle,
+                &time_extractor,
+                chunk_byte_size,
+            );
+
+            match poll_result_sender.send(Some(batch_result)) {
+                Ok(_) => work_query.waker.wake_by_ref(),
+                Err(_) => return Ok(()), // receiver disconnected, so this thread can abort
+            };
+
+            *work_query = match work_query_receiver.recv() {
+                Ok(work_query) => work_query,
+                Err(_) => return Ok(()), // sender disconnected, so there will be no new work
+            };
+        }
+
+        if poll_result_sender.send(None).is_ok() {
+            work_query.waker.wake_by_ref();
+        }
+
+        Ok(())
+    }
+
+    fn create_time_parser(
+        time_format: &Option<OgrSourceTimeFormat>,
+    ) -> Box<dyn Fn(&str) -> Result<TimeInstance> + '_> {
+        match time_format {
+            None | Some(OgrSourceTimeFormat::Iso) => Box::new(move |date: &str| {
+                let date_time = DateTime::parse_from_rfc3339(date)?;
+                Ok(date_time.timestamp_millis().into())
+            }),
+            Some(OgrSourceTimeFormat::Custom { custom_format }) => Box::new(move |date: &str| {
+                let date_time = DateTime::parse_from_str(date, &custom_format)?;
+                Ok(date_time.timestamp_millis().into())
+            }),
+            Some(OgrSourceTimeFormat::Seconds) => Box::new(move |date: &str| {
+                let date_time = DateTime::parse_from_str(date, "%C")?;
+                Ok(date_time.timestamp_millis().into())
+            }),
+        }
+    }
+
+    fn initialize_time_extractors(
+        dataset_information: &OgrSourceDataset,
+    ) -> Result<Box<dyn Fn(&Feature) -> Result<TimeInterval> + '_>> {
+        Ok(match dataset_information.time {
+            OgrSourceDatasetTimeType::None => {
+                Box::new(move |_feature: &Feature| Ok(TimeInterval::default()))
+            }
+            OgrSourceDatasetTimeType::Start => {
+                let time_start_column_name = dataset_information
+                    .columns
+                    .as_ref()
+                    .and_then(|c| c.time1.as_ref())
+                    .ok_or(Error::TimeIntervalColumnNameMissing)?;
```

Comment:
Do you have any suggestion?

Reply:
A better design of the input parameters would solve this. if `time` was an enum and the value `StartDuration` requires the time interval column specification to be present this can already fail while parsing the operator json.

Reply:
9c8d595